pkg/cluster/px/px.go

```diff
@@ -524,21 +528,28 @@ func (ops *pxClusterOps) upgradePX(newVersion string) error {
 			return nil, true, errNoPXDaemonset
 		}
 
+		pxImage := fmt.Sprintf("%s:%s", spec.PXImage, spec.PXTag)
+		ociImage := fmt.Sprintf("%s:%s", spec.OCIMonImage, spec.OCIMonTag)
+
 		for _, ds := range dss {
 			skip := false
 			logrus.Infof("Upgrading PX daemonset: [%s] %s to version: %s", ds.Namespace, ds.Name, newVersion)
 			dsCopy := ds.DeepCopy()
 			for i := 0; i < len(dsCopy.Spec.Template.Spec.Containers); i++ {
 				c := &dsCopy.Spec.Template.Spec.Containers[i]
 				if c.Name == pxDaemonsetContainerName {
-					if c.Image == newVersion {
+					oldPxImage := getEnv(c.Env, pxImageEnvKey)
+					if c.Image == ociImage && oldPxImage == pxImage {
 						logrus.Infof("Skipping upgrade of PX daemonset: [%s] %s as it is already at %s version.",
-							ds.Namespace, ds.Name, newVersion)
+							ds.Namespace, ds.Name, ociImage)
 						expectedGenerations[ds.UID] = ds.Status.ObservedGeneration
 						skip = true
 					} else {
 						expectedGenerations[ds.UID] = ds.Status.ObservedGeneration + 1
-						c.Image = newVersion
+						c.Image = ociImage
+						if spec.OCIMonTag != spec.PXTag {
```

Comment:
`if spec.OCIMonTag != spec.PXTag {`

Does this handle the case where the daemonset has PX_IMAGE=portworx/px-enteprrise:2.0.3.1

and users requests spec.OCIMonTag=2.0.3.3 and spec.PXTag=2.0.3.3?